### DISCUSSION TITLE
Use libgraphene for data types

### DISCUSSION THREAD

**Discussion by tintou** [libvips/libvips#2756] [Ideas]
## Questions, enhancements, tips, etc.

To be able to benefit from the optimizations from others, the [Graphene library](https://ebassi.github.io/graphene/) has been created and is now broadly available in distributions.

It would allow many libraries to share their data types and get rid of the many redundant implementations of points/rectangles/vectors.

**Comment by jcupitt** [comment]
Hi @tintou,

This looks interesting, though libvips has very modest needs: just 2D points and rectangles. 

Looking st the libvips sources, it's only 100 lines of C for the libvips rect functions, and making the switch would involve touching many, many lines of tested code. This doesn't seem like a useful saving to me.
